rbi/core/file.rbi

```diff
@@ -1470,7 +1470,7 @@ class File::Stat < Object
   # ```
   sig do
     params(
-        other: File::Stat,
+        other: BasicObject,
```

Comment:
could we consistently use Object or BasicObject(Given comment of trevor seems like Object is the way to go)